README.md

```diff
@@ -1 +1,113 @@
-# @moxy/react-modal
\ No newline at end of file
+# react-modal
+
+[![NPM version][npm-image]][npm-url] [![Downloads][downloads-image]][npm-url] [![Build Status][travis-image]][travis-url] [![Coverage Status][codecov-image]][codecov-url] [![Dependency status][david-dm-image]][david-dm-url] [![Dev Dependency status][david-dm-dev-image]][david-dm-dev-url]
+
+[npm-url]:https://npmjs.org/package/@moxy/react-modal
+[downloads-image]:https://img.shields.io/npm/dm/@moxy/react-modal.svg
+[npm-image]:https://img.shields.io/npm/v/@moxy/react-modal.svg
+[travis-url]:https://travis-ci.org/moxystudio/react-modal
+[travis-image]:https://img.shields.io/travis/moxystudio/react-modal/master.svg
+[codecov-url]:https://codecov.io/gh/moxystudio/react-modal
+[codecov-image]:https://img.shields.io/codecov/c/github/moxystudio/react-modal/master.svg
+[david-dm-url]:https://david-dm.org/moxystudio/react-modal
+[david-dm-image]:https://img.shields.io/david/moxystudio/react-modal.svg
+[david-dm-dev-url]:https://david-dm.org/moxystudio/react-modal?type=dev
+[david-dm-dev-image]:https://img.shields.io/david/dev/moxystudio/react-modal.svg
+
+A modal, built on top of [react-modal](https://github.com/reactjs/react-modal), that also fixes the scrollable body in iOS.
+
+## Installation
+
+```sh
+$ npm install @moxy/react-modal
+```
+
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
+
+
+## Usage
+
+First, make sure to bind modal to your appElement.
+
+```js
+import Modal from '@moxy/react-modal';
+
+Modal.setAppElement('#yourAppElement');
+```
+
+### Self controlled
+
+```js
+import React, { useRef, useCallback } from 'react';
+import Modal from '@moxy/react-modal';
+
+const MyComponent = () => {
+    const modalRef = useRef();
+
+    const handleModalOpen = useCallback(() => {
+        modalRef.current.open();
+    }, []);
+
+    return (
+        <>
+            <button onClick={ handleModalOpen }>Open</button>
+            <Modal>Modal content</Modal>
+        </>
+    );
+};
+
+export default MyComponent;
+```
+
+### Uncontrolled
+
+In this case, you have to close the modal when `onRequestClose` is called.
+
+```js
+import React, { useState, useCallback } from 'react';
+import Modal from '@moxy/react-modal';
+
+const MyComponent = () => {
+    const [isOpen, setOpen] = useState(false);
```

Comment:
Personal opinion, but i prefer setIsOpen which i think makes it clearer that this is a setter for isOpen.